### ISSUE TITLE
Failed to create a HomeKit accessory 

### ISSUE THREAD

**Issue by dasfuu** [iMicknl/ha-tahoma#508]
### Did you read the instructions?

- [X] I have read the [README](https://github.com/iMicknl/ha-tahoma/blob/master/README.md), including the [advanced debugging](https://github.com/iMicknl/ha-tahoma/blob/master/README.md#advanced) section.

### The problem

When I create a Homekit bridge in the UI and add the cover to them they dont show up in HomeKit.

### What version of this integration (ha-tahoma) has the issue?

2.5.2

### What version of Home Assistant Core has the issue?

core-2021.8.1

### Which gateway / hub do you use?

Tahoma Switch

### Device model

PositionableRollerShutter

### Device type

io:RollerShutterGenericIOComponent

### Relevant log output

```shell
2021-08-07 23:24:03 ERROR (MainThread) [homeassistant.components.homekit] Failed to create a HomeKit accessory for cover.wohnzimmer_rechts
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/homekit/__init__.py", line 628, in add_bridge_accessory
    acc = get_accessory(self.hass, self.driver, state, aid, conf)
  File "/usr/src/homeassistant/homeassistant/components/homekit/accessories.py", line 206, in get_accessory
    return TYPES[a_type](hass, driver, name, state.entity_id, aid, config)
  File "/usr/src/homeassistant/homeassistant/components/homekit/type_covers.py", line 348, in __init__
    super().__init__(
  File "/usr/src/homeassistant/homeassistant/components/homekit/type_covers.py", line 277, in __init__
    super().__init__(*args, category=category, service=service)
  File "/usr/src/homeassistant/homeassistant/components/homekit/type_covers.py", line 205, in __init__
    super().__init__(*args, category=category)
  File "/usr/src/homeassistant/homeassistant/components/homekit/accessories.py", line 250, in __init__
    firmware_revision=sw_version[:MAX_VERSION_LENGTH],
TypeError: 'NoneType' object is not subscriptable
```

**Comment by tetienne** [comment]
I'm sorry but I didn't see the link between your error and our component. In the stack trace, that's the homekit component which is involved.

**Comment by dasfuu** [comment]
To be clear: everything else works via HomeKit as intended

**Comment by iMicknl** [comment]
Thanks for reporting, @dasfuu. Did you always had this issue or only since a while?

Technically, it seems that this is an issue of the `homekit` integration since they cannot handle our firmware name. Could you have a look what is shown as 'firmware' for your device?

**Comment by dasfuu** [comment]
I only just installed Homeassistent.

The firmware seems to have a weird character in it: `5100394X22`
 The ? is probably a placeholder.

```
Rssi level
76
Core:SupportedManufacturerSettingsCommands
dead_man_up, dead_man_down, dead_man_stop, dead_man_impulse_up, dead_man_impulse_down, enter_settings_mode, save_upper_end_limit, save_lower_end_limit, stop_after_save_limit, save_settings, invert_rotation, save_my_position, delete_my_position, reset_actuator, double_power_cut, eject_from_setting_mode
Core:FirmwareRevision
5100394X22?
Core:Manufacturer
Somfy
Core:NameState
Küche
Core:PriorityLockTimerState
0
Core:StatusState
available
Core:DiscreteRSSILevelState
normal
Core:RSSILevelState
76
Core:MovingState
true
Core:ClosureState
0
Core:OpenClosedState
open
Core:Memorized1PositionState
82
Core:TargetClosureState
0
Core:ManufacturerSettingsState
-
```

**Comment by iMicknl** [comment]
I would log this bug first at Home Assistant core, mentioning the error and which firmware you have at Configuration -> Integration -> Devices -> [your device] (firmware field).

**Comment by dasfuu** [comment]
Now i know what firmware you mean.

Firmware: io:RollerShutterGenericIOComponent

**Comment by chavalaloco** [comment]
I have the same Problem with the device:
 io:HorizontalAwningIOComponent

**Comment by iMicknl** [comment]
@chavalaloco is this something which did happen since a recent update to Home Assistant?

Please link here the created bug at Home Assistant core and I will try to have a look at Home Assistant core in the coming weeks.

**Comment by dasfuu** [comment]
@iMicknl I created an issue there.
https://github.com/home-assistant/core/issues/54283

**Comment by chavalaloco** [comment]
> @chavalaloco is this something which did happen since a recent update to Home Assistant?
> 
> 
> 
> Please link here the created bug at Home Assistant core and I will try to have a look at Home Assistant core in the coming weeks.

No I first started with this addon a few days ago.

**Comment by psicot** [comment]
Hello, same issue here, with the installation of Home Assistant Core version 2021.8.x. Seems to be related to changes from Home assistant, not in this repository 😬

**Comment by iMicknl** [comment]
Resolved in Home Asssistant Core 2021.8.3, please update.
